src/software/simulation/physics/physics_world.h

```diff
@@ -34,13 +35,99 @@ class PhysicsWorld
     PhysicsWorld(const PhysicsWorld&)            = delete;
 
     /**
-     * Returns the current state of the world
-     * @return the current state of the world
+     * Returns the field in the physics world
+     *
+     * @return the field in the physics world
+     */
+    const Field getField() const;
+
+    /**
+     * Returns the state of the ball in the physics world, if it exists
+     *
+     * @return the state of the ball in the physics world, if it exists
+     */
+    const std::optional<BallState> getBallState() const;
+
+    /**
+     * Returns the states and IDs of all yellow robots in the physics world.
+     *
+     * There is no guarantee as to what order the robots are returned in.
+     *
+     * @return the states and IDs of all yellow robots in the physics world
+     */
+    const std::vector<RobotStateWithId> getYellowRobotStates() const;
+
+    /**
+     * Returns the states and IDs of all blue robots in the physics world.
+     *
+     * There is no guarantee as to what order the robots are returned in.
+     *
+     * @return the states and IDs of all blue robots in the physics world
+     */
+    const std::vector<RobotStateWithId> getBlueRobotStates() const;
+
+    /**
+     * Returns the current timestamp of the physics world
+     *
+     * @return the current timestamp of the physics world
+     */
+    const Timestamp getTimestamp() const;
+
+    /**
+     * Sets the state of the ball in the physics world. No more than 1 ball may exist
+     * in the physics world at a time. If there is no ball in the physics world, a ball
+     * is added with the given state. If a ball already exists, it's state is set to the
+     * given state.
+     *
+     * @param ball_state The new ball state
+     */
+    void setBallState(const BallState& ball_state);
```

Comment:
Just a thought, but perhaps we want to pass in the ball state as an optional so we can "remove" the ball after we add it? Not a blocking comment.

Reply:
there's a `removeBall` function for that